app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldReaction.js

```diff
@@ -23,85 +27,221 @@ const hasAuth = (id) => {
+          <span style={{ fontWeight: 'bold' }}>Element not found!</span>
+        </div>
+      );
+    }
+
+    const { edit } = this.props;
+
+    const link = (
+      <button
+        type="button"
+        style={{
+          cursor: 'pointer',
+          color: '#003366',
+          backgroundColor: 'transparent',
+          border: '1px solid #003366',
+          borderRadius: '4px',
+          margin: '5px',
+          outline: 'none',
+        }}
+        onClick={() => this.showReaction()}
+      >
+        {reaction.title()}
+      </button>
+    );
+
+    let image;
+    if (reaction.svgPath) {
+      image = <img src={reaction.svgPath} alt={reaction.title()} />;
+    }
+
+    const reactionStyle = edit ? {} : {
+      border: '1px solid #cccccc',
+      padding: '5px',
+    };
 
     return (
-      <div className="research-plan-field-reaction">
+      <div className="research-plan-field-reaction" style={reactionStyle}>
+        {link}
         <div className="image-container">
-          <img src={reaction.svgPath} alt={reaction.title()} />
-          <a role="link" tabIndex={0} onClick={() => this.showReaction()} style={{ cursor: 'pointer' }}>
-            {link}
-          </a>
+          {image}
         </div>
       </div>
     );
   }
 
   renderEdit() {
     const { connectDropTarget, isOver, canDrop } = this.props;
-    const { reaction } = this.state;
-    if (!hasAuth(reaction.id)) {
+    const { reaction, wasReactionSet } = this.state;
+
+    if (!hasAuth(reaction?.id)) {
       return noAuth(reaction);
     }
+
     let className = 'drop-target';
     if (isOver) className += ' is-over';
     if (canDrop) className += ' can-drop';
-    return connectDropTarget(<div className={className}>{reaction.id ? this.renderReaction(reaction) : 'Drop reaction here.'}</div>);
+
+    let content;
+    if (reaction?.id) {
+      content = this.renderReaction(reaction);
+    } else if (wasReactionSet) {
+      content = (
+        <div style={{ color: 'red', textAlign: 'left' }}>
+          <i className="fa fa-exclamation-triangle" aria-hidden="true" style={{ marginRight: '5px' }} />
+          <span style={{ fontWeight: 'bold' }}>Element not found!</span>
+        </div>
+      );
+    } else {
+      content = 'Drop reaction here.';
+    }
+
+    return connectDropTarget(
+      <div className={className}>
+        {content}
+      </div>
+    );
   }
 
   renderStatic() {
-    const { reaction } = this.state;
-    return reaction.id ? this.renderReaction(reaction) : '';
+    const { reaction, wasReactionSet } = this.state;
+
+    let content;
+    if (reaction?.id) {
+      content = this.renderReaction(reaction);
+    } else if (wasReactionSet) {
+      content = (
+        <div style={{ color: 'red', textAlign: 'left' }}>
+          <i className="fa fa-exclamation-triangle" aria-hidden="true" style={{ marginRight: '5px' }} />
+          <span style={{ fontWeight: 'bold' }}>Element not found!</span>
+        </div>
```

Comment:
This div block appears multiple times (lines 147-150 & lines 211-214), could it be refactored into its own function, something like `renderElementError()` ?

Reply:
Fixed by: [776bbe9](https://github.com/ComPlat/chemotion_ELN/pull/1452/commits/776bbe91e314ea56c75cff3c481fa22e0319d56c)